src-vue/src/views/ModsView.vue

```diff
@@ -69,6 +77,30 @@ export default defineComponent({
 
             this.global_load_indicator = false;
             return true;
+        },
+        async deleteMod(mod: NorthstarMod) {
+            let game_install = {
+                game_path: this.$store.state.game_path,
+                install_type: this.$store.state.install_type
+            } as GameInstall;
+            await invoke("delete_northstar_mod_caller", { gameInstall: game_install, nsmodName: mod.name })
+                .then((message) => {
+                    // Just a visual indicator that it worked
+                    ElNotification({
+                        title: `Success deleting ${mod.name}`,
+                        type: 'success',
+                        position: 'bottom-right'
+                    });
+                })
+                .catch((error) => {
+                    ElNotification({
+                        title: 'Error',
+                        message: error,
+                        type: 'error',
+                        position: 'bottom-right'
+                    });
+                });
+            this.$store.commit('loadInstalledMods');
```

Comment:
Since `invoke` call is asynchronous, `$store.commit` can be invoked before `.then` or `.catch`; it should be in the `.finally` callback (as I suppose you want to reload mods list after mod deletion).

Reply:
Uh could you give me a short example how exactly it should look like? Not sure I fully understand ^^"

Reply:
```javascript
await promise("blabla")
    .then(() => {
        // get executed if promise went well
    })
    .catch((err) => {
        // get executed if promised threw an error
    })
    .finally(() => {
        // get executed no matter promise result
    });
```

https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Global_Objects/Promise/finally

Reply:
Oh wait, so you just meant:

```typescript
await invoke("delete_northstar_mod", { gameInstall: game_install, nsmodName: mod.name })
    .then((message) => {
        // Just a visual indicator that it worked
        ElNotification({
            title: `Success deleting ${mod.name}`,
            type: 'success',
            position: 'bottom-right'
        });
    })
    .catch((error) => {
        ElNotification({
            title: 'Error',
            message: error,
            type: 'error',
            position: 'bottom-right'
        });
    })
    .finally(() => {
        this.$store.commit('loadInstalledMods');
    });
```
?

Reply:
Yup :)

Reply:
I thought the `await` should prevent re-ordering but I guess not? Anyway, changed in d0a8c249774260934e65c766d1ac34205ffc93b2 ^^